llvm/tools/sycl-post-link/ModuleSplitter.h

```diff
@@ -67,6 +69,7 @@ struct EntryPointGroup {
                          : SyclEsimdSplitStatus::SYCL_AND_ESIMD;
       Res.UsesLargeGRF = UsesLargeGRF || Other.UsesLargeGRF;
       // Scope remains global
+      // OptLevel is expected to be the same for both merging EPGs
```

Comment:
We only ever merge two `ModuleDesc`s together after splitting a bigger module into two by ESIMD splitter. Therefore, we can assume that `OptLevel` is the same for both of them. I would add an `assert` here instead of a comment.